dbt/models/iasworld/schema/iasworld.dweldat.yml

```diff
@@ -403,22 +403,39 @@ sources:
           - name: fixhalf
             description: '{{ doc("shared_column_char_hbath") }}'
             tests:
-              - not_null:
-                  name: iasworld_dweldat_fixhalf_not_null
+              - accepted_range:
+                  name: iasworld_dweldat_fixhalf_gte_0
+                  min_value: 0
                   additional_select_columns: *select-columns
                   config: *unique-conditions
                   meta:
                     description: >
-                      fixhalf (Number of Half Baths) should not be null
+                      fixhalf (Number of Half Baths) should be >= 0
               - accepted_range:
-                  name: iasworld_dweldat_fixhalf_between_0_and_6
+                  name: iasworld_dweldat_fixhalf_matches_number_of_units
                   min_value: 0
-                  max_value: 6
-                  additional_select_columns: *select-columns
+                  max_value: |
+                    CASE
+                      WHEN user14 IS NULL OR user14 = '0' OR user14 = '6' THEN 5
+                      WHEN user14 = '1' THEN 10
+                      WHEN user14 = '2' THEN 15
+                      WHEN user14 = '3' THEN 20
+                      WHEN user14 = '4' THEN 25
+                      WHEN user14 = '5' THEN 30
+                      ELSE 6
+                    END
+                  additional_select_columns: &select-columns-with-num-units
+                    - taxyr
+                    - parid
+                    - card
+                    - who
+                    - wen
+                    - user14
                   config: *unique-conditions
                   meta:
                     description: >
-                      fixhalf (Number of Half Baths) should be between 0 and 6
+                      fixhalf (Number of Half Baths) should be <= 5 times the
+                      number of units (user14)
```

Comment:
I spot-checked these results and the heuristic seemed to be performing reasonably well to me, but you may want to take a look at the spreadsheet output yourself as well.